src/main/serializer.c

```diff
@@ -193,11 +185,10 @@ void set_as_bytes(as_bytes **bytes, uint8_t *bytes_string,
 CLEANUP:
 
     if (error_p->code != AEROSPIKE_OK) {
-        PyObject *py_err = NULL;
-        error_to_pyobject(error_p, &py_err);
-        PyObject *exception_type = raise_exception(error_p);
-        PyErr_SetObject(exception_type, py_err);
-        Py_DECREF(py_err);
+        // TODO: when exception is detected in this method
+        // the parent method may also raise an exception
+        // is this intended behavior?
+        raise_exception(error_p);
```

Comment:
I think the child should raise an exception and everything should return.

Reply:
This might be a breaking change if the application is checking the file and line number where the error happened.